### PR TITLE
Scripting: Remove search template actions

### PR DESCRIPTION
**Pull request by rjernst**
The dedicated search template put/get/delete actions are deprecated in
5.6. This commit removes them from 6.0.

### CODE REVIEW

File: docs/reference/migration/migrate_6_0/scripting.asciidoc [hunk 1]
```diff
@@ -33,4 +33,8 @@ The `lang` variable can no longer be specified as part of a request that uses a
 script otherwise an error will occur.  Note that a request using a stored script is
 different from a request that puts a stored script.  The language of the script has
 already been stored as part of the cluster state and an `id` is sufficient to access
-all of the information necessary to execute a stored script.
\ No newline at end of file
+all of the information necessary to execute a stored script.
+
+==== Stored template apis removed
```
Comment (dakrone) [inline docs/reference/migration/migrate_6_0/scripting.asciidoc]: Whoa nelly, this should definitely read `Stored *search* template APIs removed`. I had a double-take that we were removing index templates when I first read this.

File: docs/reference/migration/migrate_6_0/scripting.asciidoc [hunk 2]
```diff
@@ -33,4 +33,8 @@ The `lang` variable can no longer be specified as part of a request that uses a
...
+
+The PUT, GET and DELETE _template apis have been removed. Store templates with the stored scripts apis instead.
```
Comment (dakrone) [inline docs/reference/migration/migrate_6_0/scripting.asciidoc]: Again, this is not the `/_template` APIs, this is `/_search/template` APIs.

I do think it'd be extremely beneficial to have a "if you did this, now do this"
part in these docs as a quick example

Here's an example if you want to use it:

If you previously did:

```json
PUT /_search/template/my_template
{
  "query": {
    "match": {
      "f1": "{{f1}}"
    }
  }
}
```

You should do now do

```json
PUT /_scripts/my_template
{
  "script": {
    "lang": "mustache",
    "source": {
      "query": {
        "match": {
          "f1": "{{f1}}"
        }
      }
    }
  }
}
```

**Review by dakrone — CHANGES_REQUESTED** [review]
I think we definitely need to be clearer in the breaking docs for this (code-wise this looks fine)

**Comment by rjernst** [comment]
Thanks @dakrone! I often get confused with the template vs search template. I've fixed the docs and added the examples.

**Review by dakrone — APPROVED** [review]
LGTM, thanks!

**Pull request merged**